src/core/mcs.rs

```diff
@@ -226,6 +226,7 @@ impl<S: Read + Write> Client<S> {
         screen_height: u16,
         keyboard_layout: KeyboardLayout,
         client_name: String,
+        static_channels: &Vec<String>,
```

Comment:
Nit: `&Vec<T>` is almost never what you want.  Prefer using `&[T]` instead.  Its more performant (slices are direct non-owning pointers to regions of contiguous memory, whereas `&Vec` is a non-owning pointer to an owning pointer), and more flexible (slices can come from other types than just `Vec`s).

(this nit applies to a few places in the pr)